eng/common/TestResources/New-TestResources.ps1

```diff
@@ -286,7 +286,21 @@ foreach ($templateFile in $templateFiles) {
         Log "Persist the following environment variables based on your detected shell ($shell):`n"
     }
 
-    $deploymentOutputs = @{}
+    $serviceDirectoryPrefix = $ServiceDirectory.ToUpperInvariant()
+
+    $context = Get-AzContext;
+
+    # Add default values
+    $deploymentOutputs = @{
+        "$($serviceDirectoryPrefix)_CLIENT_ID" = $TestApplicationId;
```

Comment:
Is there a reason these have to be specific to the service? It seems like most of these should be general and shared.

Reply:
I want to eventually get to a place where all live tests "environments" can be independent from each other and be running in parallel. I know CLIENT_ID is often common but resource group is often not.

Reply:
Thinking about this more, I think we should put "AZURE_" as a fixed prefix before this. If we're going to be asking a user to define all these values, it's going to be harder to clean up. If there's a common prefix, they could do something like the following to clean them up quickly, easily, and *completely*. We don't want to inadvertently encourage them to leave secrets lying around longer than desired.

```powershell
dir env:AZURE* | del
```

Reply:
It gives them the option to completely clean up and it's a simple change.

Reply:
It's not completely, test-resource.json output variables are not scoped.

Reply:
They are in my original test-resources.json (for Key Vault) and any that originally copied it. From a quick PS query, I see you're right, though:

```
APPCONFIGURATION_CONNECTION_STRING
APPCONFIGURATION_ENDPOINT_STRING
AZURE_CLIENT_ID
AZURE_CLIENT_ID
AZURE_CLIENT_ID
AZURE_CLIENT_ID
AZURE_CLIENT_ID
AZURE_CLIENT_ID
AZURE_CLIENT_SECRET
AZURE_CLIENT_SECRET
AZURE_CLIENT_SECRET
AZURE_CLIENT_SECRET
AZURE_CLIENT_SECRET
AZURE_CLIENT_SECRET
AZURE_KEYVAULT_URL
AZURE_KEYVAULT_URL
AZURE_LOCATION
AZURE_RESOURCE_GROUP
AZURE_SEARCH_STORAGE_KEY
AZURE_SEARCH_STORAGE_NAME
AZURE_SUBSCRIPTION_ID
AZURE_TENANT_ID
AZURE_TENANT_ID
AZURE_TENANT_ID
AZURE_TENANT_ID
AZURE_TENANT_ID
AZURE_TENANT_ID
BLOB_STORAGE_ACCOUNT_KEY
BLOB_STORAGE_ACCOUNT_KEY
BLOB_STORAGE_ACCOUNT_NAME
BLOB_STORAGE_ACCOUNT_NAME
DATALAKE_STORAGE_ACCOUNT_KEY
DATALAKE_STORAGE_ACCOUNT_NAME
EVENT_HUBS_CLIENT
EVENT_HUBS_RESOURCEGROUP
EVENT_HUBS_SECRET
EVENT_HUBS_SUBSCRIPTION
EVENT_HUBS_TENANT
FORM_RECOGNIZER_API_KEY
FORM_RECOGNIZER_BLOB_CONTAINER_SAS_URL
FORM_RECOGNIZER_ENDPOINT
PREMIUM_STORAGE_ACCOUNT_KEY
PREMIUM_STORAGE_ACCOUNT_NAME
PRIMARY_STORAGE_ACCOUNT_KEY
PRIMARY_STORAGE_ACCOUNT_NAME
SECONDARY_STORAGE_ACCOUNT_KEY
SECONDARY_STORAGE_ACCOUNT_NAME
SERVICE_BUS_CLIENT
SERVICE_BUS_RESOURCEGROUP
SERVICE_BUS_SECRET
SERVICE_BUS_SUBSCRIPTION
SERVICE_BUS_TENANT
TEXT_ANALYTICS_API_KEY
TEXT_ANALYTICS_ENDPOINT
```

I will propose to address this differently. I think it's risky to be splattering potentially sensitive variables like that.

Reply:
Risky how? We are already doing it.